src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -88,6 +88,36 @@ ov::Tensor tensor_from_numpy(py::array& array, bool shared_memory) {
     return tensor;
 }
 
+ov::PartialShape partial_shape_from_list(const py::list& shape) {
+    using value_type = ov::Dimension::value_type;
+    std::vector<ov::Dimension> dims;
+    for (py::handle dim : shape) {
+        if (py::isinstance<py::int_>(dim)) {
+            dims.push_back(ov::Dimension(dim.cast<value_type>()));
+        } else if (py::isinstance<ov::Dimension>(dim)) {
+            dims.push_back(dim.cast<ov::Dimension>());
+        } else if (py::isinstance<py::list>(dim) || py::isinstance<py::tuple>(dim)) {
+            py::list bounded_dim = dim.cast<py::list>();
+            py::type_error error("Incorrect values in shape. Can't create bounded dimension from " +
+                                 std::string(dim.str()) + ".");
```

Comment:
Provide more information about allowed types and what's going wrong. For example, "two elements are expected in tuple (lower, upper) for dynamic dimensions, N provided instead" or something similar. Add more guidance, and not only in this place.

Reply:
done